### PR TITLE
add language page

### PR DESCRIPTION
**Pull request by belljohan3**

### CODE REVIEW

**Comment by ivanslabbert** [comment]
The modal still allows saving languages with no groups set when editing:
image
This might be because the language has a group membership that is not visible to the user? In such a case I think we need to disable editing as the user needs to have edit access to all the groups before he is allowed to edit the language.

**Review by DirkDevelopmentFreelance — COMMENTED** [review]
It looks very nice and clean! Nice work! We just need to add a indication for offline changes like in the ContentOverview/EditContent

**Comment by belljohan3** [comment]
- [ ] add the Offline Changes
- [ ] Check permission (@ivanslabbert)

File: cms/src/components/languages/CreateOrEditLanguageModal.vue [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+// Check if we are in edit mode (if a language is passed)
+const isEditMode = computed(() => !!props.language);
+
+// New language or edited language object
+const newLanguage = ref<LanguageDto>({
+    _id: db.uuid(), // Generate new ID for create mode
+    name: "",
+    languageCode: "",
+    memberOf: ["group-languages"],
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModal.vue:30]: We should not hardcode a default group as this group might not exist in a production system.

File: cms/src/components/languages/CreateOrEditLanguageModal.vue [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        newLanguage.value.name.trim() !== "" &&
+        newLanguage.value.languageCode.trim() !== "" &&
+        newLanguage.value.memberOf.length > 0
+    );
+};
+
+// Function to close the modal
+const closeModal = () => {
+    emit("close");
+};
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModal.vue:98]: Maybe cleaner to do the close modal code inline in html?

File: cms/src/components/languages/CreateOrEditLanguageModel.spec.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        accessMap.value = fullAccessToAllContentMap;
+    });
+
+    afterEach(async () => {
+        // Clear the database after each test
+        await db.docs.clear();
+        await db.localChanges.clear();
+    });
+
+    describe("on create mode", () => {
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModel.spec.ts:30]: It is a bit confusing with the naming. Perhaps it should just read "Create mode"?

File: cms/src/components/languages/CreateOrEditLanguageModel.spec.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        it("display modal on create mode", async () => {
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModel.spec.ts:31]: "can display a modal in create mode"

File: cms/src/components/languages/CreateOrEditLanguageModel.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                .element as HTMLInputElement;
+
+            expect(wrapper.html()).toContain("Create New Language");
+            expect(inputLanguageName.value).toBe("");
+
+            // check if the button has the right text
+            expect(wrapper.html()).toContain("Create");
+        });
+
+        it("enable save button if form fields are filled", async () => {
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModel.spec.ts:48]: "can enable..."

File: cms/src/components/languages/CreateOrEditLanguageModel.spec.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            await wrapper.find("[name='languageName']").setValue("Afrikaans");
+            await wrapper.find("[name='languageCode']").setValue("afr");
+
+            // Assert that the save button is disabled
+            const saveButton = wrapper.find("[data-test='save-button']");
+            expect(saveButton.attributes("disabled")).toBeUndefined();
+        });
+    });
+
+    describe("on edit mode", () => {
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModel.spec.ts:65]: "edit mode"

File: cms/src/components/languages/CreateOrEditLanguageModel.spec.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        it("displays modal on edit mode", async () => {
```
Comment (ivanslabbert) [inline cms/src/components/languages/CreateOrEditLanguageModel.spec.ts:66]: "can display the modal in edit mode"

**Review by ivanslabbert — APPROVED** [review]
Hi @belljohan3 

Approved the PR with a few comments. Also noticed that offline changes are not shown, but this is not a priority as we will sometime in the future remove the offline editing functionality from core functions such as group and language editing.

**Pull request merged**